sql/rowexec: TestJoinReaderDiskSpill failed

sql/rowexec.TestJoinReaderDiskSpill [failed](https://mesolite.cluster.engflow.com/invocations/default/d093d866-6b9d-477b-a3b9-a6b2a39b1f93?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3Jvd2V4ZWM6cm93ZXhlY190ZXN0) on master @ [e6145612f7bfa33ac19c69e288763cbf4bffe158](https://github.com/cockroachdb/cockroach/commits/e6145612f7bfa33ac19c69e288763cbf4bffe158):


```
=== RUN   TestJoinReaderDiskSpill
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestJoinReaderDiskSpill2950516679
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:183: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    joinreader_test.go:1327: 
        	Error Trace:	pkg/sql/rowexec/joinreader_test.go:1327
        	Error:      	Should be true
        	Test:       	TestJoinReaderDiskSpill
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestJoinReaderDiskSpill2950516679
--- FAIL: TestJoinReaderDiskSpill (1.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJoinReaderDiskSpill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58744

sql/rowexec.TestJoinReaderDiskSpill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20984082?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20984082?buildTab=artifacts#/) on master @ [81f046c5eb268fcf87c8c34fa099dc291cdf24aa](https://github.com/cockroachdb/cockroach/commits/81f046c5eb268fcf87c8c34fa099dc291cdf24aa):


```
=== RUN   TestJoinReaderDiskSpill
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill3787951157
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:183: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    joinreader_test.go:1327: 
        	Error Trace:	pkg/sql/rowexec/joinreader_test.go:1327
        	Error:      	Should be true
        	Test:       	TestJoinReaderDiskSpill
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill3787951157
--- FAIL: TestJoinReaderDiskSpill (1.47s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJoinReaderDiskSpill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/rowexec.TestJoinReaderDiskSpill [failed](https://mesolite.cluster.engflow.com/invocations/default/69d28a93-5e40-4eea-bcde-115381f90ad3?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3Jvd2V4ZWM6cm93ZXhlY190ZXN0) on master @ [4edd331847c335b46295b1e053715c052b75eb13](https://github.com/cockroachdb/cockroach/commits/4edd331847c335b46295b1e053715c052b75eb13):


```
=== RUN   TestJoinReaderDiskSpill
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestJoinReaderDiskSpill4282163090
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    joinreader_test.go:1327: 
        	Error Trace:	pkg/sql/rowexec/joinreader_test.go:1327
        	Error:      	Should be true
        	Test:       	TestJoinReaderDiskSpill
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestJoinReaderDiskSpill4282163090
--- FAIL: TestJoinReaderDiskSpill (4.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJoinReaderDiskSpill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/rowexec.TestJoinReaderDiskSpill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20986086?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20986086?buildTab=artifacts#/) on master @ [9cdc6d65fea7b18236ab07ef5a377c1492268cc6](https://github.com/cockroachdb/cockroach/commits/9cdc6d65fea7b18236ab07ef5a377c1492268cc6):


```
=== RUN   TestJoinReaderDiskSpill
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill1683956622
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    joinreader_test.go:1327: 
        	Error Trace:	pkg/sql/rowexec/joinreader_test.go:1327
        	Error:      	Should be true
        	Test:       	TestJoinReaderDiskSpill
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill1683956622
--- FAIL: TestJoinReaderDiskSpill (0.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJoinReaderDiskSpill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/rowexec.TestJoinReaderDiskSpill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20990591?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20990591?buildTab=artifacts#/) on master @ [975f0b5107ebc3e29711d010b58a65f1e9a38b52](https://github.com/cockroachdb/cockroach/commits/975f0b5107ebc3e29711d010b58a65f1e9a38b52):


```
=== RUN   TestJoinReaderDiskSpill
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill3504458477
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:183: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    joinreader_test.go:1327: 
        	Error Trace:	pkg/sql/rowexec/joinreader_test.go:1327
        	Error:      	Should be true
        	Test:       	TestJoinReaderDiskSpill
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill3504458477
--- FAIL: TestJoinReaderDiskSpill (1.67s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJoinReaderDiskSpill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/rowexec.TestJoinReaderDiskSpill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20991905?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20991905?buildTab=artifacts#/) on master @ [ceac981045c8256787347794175e2f9372bc8ba5](https://github.com/cockroachdb/cockroach/commits/ceac981045c8256787347794175e2f9372bc8ba5):


```
=== RUN   TestJoinReaderDiskSpill
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/d8701f61058810787832e356cd52137f/logTestJoinReaderDiskSpill4240784229
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:183: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    joinreader_test.go:1327: 
        	Error Trace:	pkg/sql/rowexec/joinreader_test.go:1327
        	Error:      	Should be true
        	Test:       	TestJoinReaderDiskSpill
    panic.go:615: -- test log scope end --
--- FAIL: TestJoinReaderDiskSpill (2.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJoinReaderDiskSpill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/rowexec/joinreader_test.go
+++ b/pkg/sql/rowexec/joinreader_test.go
@@ -1218,8 +1218,6 @@ func TestJoinReaderDiskSpill(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 161212)
-
 	ctx := context.Background()
 
 	srv, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
@@ -1230,7 +1228,7 @@ func TestJoinReaderDiskSpill(t *testing.T) {
 	// a disk spill.
 	key := 0
 	stringColVal := "0123456789"
-	numRows := 100
+	numRows := 1000
 	if _, err := sqlDB.Exec(`
 CREATE DATABASE test;
 CREATE TABLE test.t (a INT, s STRING, INDEX (a, s))`); err != nil {

